### PR TITLE
Update container source documentation with `template` field

### PR DESCRIPTION
**Pull request by jingweno**
Fixes https://github.com/knative/eventing/issues/1348

## Proposed Changes

https://github.com/knative/eventing/pull/1321 introduces `PodTemplateSpec` to `ContainerSourceSpec`. This feature will be out in 0.7.x. Updating documentation to reflect the `template` field.

### CODE REVIEW

**Comment by jingweno** [comment]
@n3wscott I'm unsure why the tests failed. Any pointers?

**Comment by samodell** [comment]
Thanks, @jingweno !
/approve

**Comment by jingweno** [comment]
According to the [build log](https://00e9e64bace2b630e13c2fa1ba1619a38351f572e3158165d2-apidata.googleusercontent.com/download/storage/v1/b/knative-prow/o/pr-logs%2Fpull%2Fknative_docs%2F1491%2Fpull-knative-docs-integration-tests%2F1141477278069297152%2Fbuild-log.txt?qk=AD5uMEuFDELNtEszr6N7EQXdUkX0tvhuWbP3uIdvss0RtjyBu4-QgilvLMXXYflKy6V2VxkT9PJw5Kw1qrsViZ_wfAmEnHm4p12Tp9KpdIpCa-prGpCiSrzovKQa_QiUXbNsIRpsmWhiIYfD-kSR1P5ceLoNHS5qbRrmhA7B2s1vU1x5pp0F9ke2wWXdTGQCPZP2pYV1Pd4cnqyMHSE9yh6vqS6aGTsjYIrlX3KWt17wInuMoLIY7nj8mCsYJYpZRHmY6V1RDlpf-8THK20oUR6f8Yy3XJZE4ID2FwNfolj8FAw7B-GaicCwEklWK5ECxJ9jCcHVx8GVS6YTRT80btWwY1epwiPKu0gfy9oX6jFImv-Lm7oouQX1zghSgnSo3xztA2m0u4gxR08iZ0mOUOZ_fyrVT2Ct4eK21IWkFaol0gqI2a8bpfa7dUXqZrSyN-M_vDBHXxcA2pMFYcFUIU8HAzhQznJk8LXNpzbpcbHzrFJDiIsmjmHlVTtGoI99BZI9D0VYeBxPolTNC8qPqmN790cWXFyluZSMm8-A8ApG_wdCQsklfL1lR3bfujLXcuFgQKsCMbmcKxaJnu8PTjhYLazb2j1lj-w1HhvMHwEl3oJgQhLKNoJsGftZbmbJQ8CBv1-UkLSN7kfSOFayC4jGdNzOTwvn53dbDqQvIeEhnqCY6m68k1-5H6dy1NTabHWoHnLUcn7irAvcEwGfMqT--VOHBPIhemAJY4ZHVWwssSTEkfyUZ0b49BcW9YlX_mwCq54llrw1QKl0D4w3K_RKXD4jdQGmhgWdp2xNwzvLCeeOTprr1llOI5tWUoYj803tWLeebQpZSCWR2NEeKhsJsiSXB4jRohZ0KBsHQ51cTba8JgL8GCiLhiXuBThLPsZoo_lF-hnGJIwCoFlJ0mfwNOgCUa-0dQ), the e2e tests failed at https://github.com/knative/docs/blob/master/test/e2e/sampleapptestbase.go#L111 due to tests running to long: `Test killed with quit: ran too long (10m0s)`. I have no idea why `docker build` would fail with a doc change.

**Comment by samodell** [comment]
Yeah, it's definitely an issue with the tests. Hey @chaodaiG , could you take a look at this failure?

**Comment by chaodaiG** [comment]
Master branch looks fine to me, and these change shouldn't break test
/test pull-knative-docs-integration-tests

**Comment by Fredy-Z** [comment]
/hold
@jingweno https://github.com/knative/eventing/issues/1395 also needs your attention.
Can you update https://github.com/knative/docs/blob/master/docs/eventing/broker-trigger.md as well after https://github.com/knative/eventing-contrib/pull/469 is in?

**Comment by jingweno** [comment]
@Fredy-Z:

> Can you update https://github.com/knative/docs/blob/master/docs/eventing/broker-trigger.md in this PR as well after knative/eventing-contrib#469 is in?

I think the broker/trigger doc fix is better to be in another PR because this one is about updating the doc for the new `template` field of container source. I saw that you have already another PR to fix the broker/trigger doc: https://github.com/knative/docs/pull/1496

**Comment by Fredy-Z** [comment]
> @Fredy-Z:
> 
> > Can you update https://github.com/knative/docs/blob/master/docs/eventing/broker-trigger.md in this PR as well after [knative/eventing-contrib#469](https://github.com/knative/eventing-contrib/pull/469) is in?
> 
> I think the broker/trigger doc fix is better to be in another PR because this one is about updating the doc for the new `template` field of container source. And I saw that you already have a PR to fix the broker/trigger doc: #1496

My PR is to update docs of release-0.6 branch.
The master branch also needs to be updated, and since this broker/trigger doc uses ContainerSource as a demo (see `Knative Source` part of https://github.com/knative/docs/blob/master/docs/eventing/broker-trigger.md), it also needs to be updated by using the new `template` field of ContainerSource.

File: docs/eventing/samples/container-source/README.md [hunk 1]
```diff
@@ -71,18 +71,21 @@ kind: ContainerSource
 metadata:
   name: test-heartbeats
 spec:
-  image: <heartbeats_image_uri>
+  template:
+    spec:
+      containers:
+        - image: <heartbeats_image_uri>
```
Comment (Fredy-Z) [inline docs/eventing/samples/container-source/README.md]: Seems container `name` is missed here, this yaml template will result in an error.

File: docs/eventing/samples/container-source/heartbeats-source.yaml [hunk 2]
```diff
@@ -3,17 +3,20 @@ kind: ContainerSource
...
-  # e.g. gcr.io/[gcloud-project]/github.com/knative/eventing-contrib/cmd/heartbeats
-  image: index.docker.io/daisyycguo/heartbeats-6790335e994243a8d3f53b967cdd6398
-  args: 
-    - --period=1
+  template:
+    spec:
+      containers:
+        # This corresponds to a heartbeats image uri you build and publish,
+        # e.g. gcr.io/[gcloud-project]/github.com/knative/eventing-contrib/cmd/heartbeats
+        - image: index.docker.io/jingweno/heartbeats-6410d29c99b6ec4e0dd2fb3e6e31829e
```
Comment (Fredy-Z) [inline docs/eventing/samples/container-source/heartbeats-source.yaml]: Ditto.

File: docs/eventing/samples/container-source/README.md (hunk 1 above)
Comment (jingweno) [reply docs/eventing/samples/container-source/README.md]: @Fredy-Z Good catch. Fixed.

File: docs/eventing/samples/container-source/heartbeats-source.yaml (hunk 2 above)
Comment (jingweno) [reply docs/eventing/samples/container-source/heartbeats-source.yaml]: Fixed

**Comment by jingweno** [comment]
@Fredy-Z:

> My PR is to update docs of release-0.6 branch.
The master branch also needs to be updated, and since this broker/trigger doc uses ContainerSource as a demo (see Knative Source part of https://github.com/knative/docs/blob/master/docs/eventing/broker-trigger.md), it also needs to be updated by using the new template field of ContainerSource.

Gotcha. I saw that you also fixed master: https://github.com/knative/docs/pull/1504. I should have addressed all feedback in this PR.

**Pull request merged**